### PR TITLE
`cudf::rolling` `ROW_NUMBER` support for `decimal32` and `decimal64`

### PR DESCRIPTION
**Pull request by codereport**
This PR adds support for `cudf::rolling` for the `ROW_NUMBER` option for `decimal32` and `decimal64`. It also clarifies the documentation.

### CODE REVIEW

File: cpp/tests/rolling/rolling_test.cpp [hunk 1]
```diff
@@ -975,23 +975,30 @@ TYPED_TEST(FixedPointTests, MinMaxCountLagLead)
...
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_max, max->view());
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_lag, lag->view());
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_lead, lead->view());
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_count_val, valid->view());
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_count_all, all->view());
-  // CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_rowno, rowno->view());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_rowno, rowno->view());
+
+  // ROW_NUMBER will always return row 1 if the preceding window is set to a constant 1
+  for (int following = 1; following < 5; ++following) {
```
Comment (davidwendt) [inline cpp/tests/rolling/rolling_test.cpp:998]: What is the significance of `5` here? Is it `input.size()-1`?
I'm not very familiar with rolling.

Comment (codereport) [reply cpp/tests/rolling/rolling_test.cpp:998]: 5 is arbitrary. could be anything less <= input size.

**Review by davidwendt — APPROVED** [review]
Looks good to me.

File: cpp/include/cudf/aggregation.hpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ class aggregation {
     ARGMIN,          ///< Index of min element
     NUNIQUE,         ///< count number of unique elements
     NTH_ELEMENT,     ///< get the nth element
-    ROW_NUMBER,      ///< get row-number of element
+    ROW_NUMBER,      ///< get row-number of current index (relative to rolling window)
```
Comment (mythrocks) [inline cpp/include/cudf/aggregation.hpp:76]: Thank you for fixing this, @codereport.

**Review by devavret — APPROVED** [review]
So, no changes needed? Nice

**Pull request merged**